testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/ResultAssertion.java

```diff
@@ -146,6 +148,9 @@ public void validate(String output, String displayName) {
                 i++;
             } else if (isDeprecationMessageInHelpDescription(line)) {
                 i++;
+            } else if (getNumLinesFromDaemonLogOutput(line) != -1) {
+                // Daemon logs may duplicate deprecation warnings
+                i += getNumLinesFromDaemonLogOutput(line);
```

Comment:
🤔 Can we refactor this to only count the daemon output lines once?

Reply:
I thought about that but given the way the if statement chain is written it would be more pain that it’s worth imo.

in any case the number of lines is only calculated when we actually detect the daemon output, which itself is rare

Reply:
It's a little non-standard, but you could do:

```
int daemonLines;

if //...
else if ((daemonLines = getNumLinesFromDaemonLogOutput(line)) != -1) {
  i += daemonLines;
} else // ...
```

Reply:
I would personally stick to being standard here, though that looks like it would work we’re trading readability for negligible performance